integration_tests/twodut_oneotg_tests/bgp_triggered_gue/bgp_triggered_gue_test.go

```diff
@@ -0,0 +1,624 @@
+			WithOperationType(constants.Add).
+			AsResult(),
+		fluent.OperationResult().
+			WithNextHopOperation(nhIndex2).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			WithOperationType(constants.Add).
+			AsResult(),
+		fluent.OperationResult().
+			WithNextHopGroupOperation(nhgIndex2).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			WithOperationType(constants.Add).
+			AsResult(),
+		fluent.OperationResult().
+			WithIPv4Operation(ateDstNetCIDR2).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			WithOperationType(constants.Add).
+			AsResult(),
+		fluent.OperationResult().
+			WithNextHopOperation(nhIndex3).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			WithOperationType(constants.Add).
+			AsResult(),
+		fluent.OperationResult().
+			WithNextHopGroupOperation(nhgIndex3).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			WithOperationType(constants.Add).
+			AsResult(),
+		fluent.OperationResult().
+			WithIPv4Operation(ateDstNetCIDR3).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			WithOperationType(constants.Add).
+			AsResult(),
+	}
+
+	for _, wantResult := range wantOperationResultsDUT1 {
+		chk.HasResult(t, c.Results(t), wantResult, chk.IgnoreOperationID())
+	}
+	for _, wantResult := range wantOperationResultsDUT2 {
+		chk.HasResult(t, c2.Results(t), wantResult, chk.IgnoreOperationID())
+	}
+
+	// Start a new BGP session that should exchange the necessary gRIBI
+	// route that recursively resolves and thus enables traffic to flow.
+	dutConf := bgpWithNbr(dutAS, dutPort3.IPv4, &oc.NetworkInstance_Protocol_Bgp_Neighbor{
+		PeerAs:          ygot.Uint32(dut2AS),
+		NeighborAddress: ygot.String(dut2Port2.IPv4),
+	})
+	dut2Conf := bgpWithNbr(dut2AS, dut2Port2.IPv4, &oc.NetworkInstance_Protocol_Bgp_Neighbor{
+		PeerAs:          ygot.Uint32(dutAS),
+		NeighborAddress: ygot.String(dutPort3.IPv4),
+	})
+	gnmi.Replace(t, dut, bgpPath.Config(), dutConf)
+	gnmi.Replace(t, dut2, bgpPath.Config(), dut2Conf)
+
+	nbrPath := bgpPath.Neighbor(dut2Port2.IPv4)
+	gnmi.Await(t, dut, nbrPath.SessionState().State(), 60*time.Second, oc.Bgp_Neighbor_SessionState_ESTABLISHED)
+
+	testTrafficAndPrintCapture := func(t *testing.T, title string, startingIP string, encapFields *EncapFields) {
+		t.Helper()
+		t.Logf("testing traffic %s (starting IP %s)", title, startingIP)
+		otg := ate.OTG()
+		otg.StartCapture(t, atePort2.Name)
+
+		if loss := testTraffic(t, ate, ateTop, atePort1, atePort2, startingIP); loss > 1 {
+			t.Errorf("Loss: got %g, want <= 1", loss)
+		}
+
+		otg.StopCapture(t, atePort2.Name)
+
+		captureBytes := ate.OTG().FetchCapture(t, atePort2.Name)
+
+		f, err := os.CreateTemp(".", "pcap")
+		if err != nil {
+			t.Fatalf("ERROR: Could not create temporary pcap file: %v\n", err)
+		}
+		defer os.Remove(f.Name())
+
+		if _, err := f.Write(captureBytes); err != nil {
+			t.Fatalf("ERROR: Could not write bytes to pcap file: %v\n", err)
+		}
+		f.Close()
+
+		handleRead, err := pcap.OpenOffline(f.Name())
```

Comment:
use https://pkg.go.dev/github.com/google/gopacket/pcapgo instead, it doesn't rely on libpcap being install on the system